core/src/forest/ForestOptions.cpp

```diff
@@ -76,8 +76,29 @@ ForestOptions::ForestOptions(uint num_trees,
                              uint random_seed,
                              const std::vector<size_t>& sample_clusters,
                              uint samples_per_cluster,
-                             uint blocklength) {
-  // TODO
+                             uint blocklength) :
+    ci_group_size(ci_group_size),
+    sample_fraction(sample_fraction),
+    tree_options(mtry, min_node_size, honesty, honesty_fraction, honesty_prune_leaves, alpha, imbalance_penalty),
+    sampling_options(samples_per_cluster, sample_clusters, blocklength){
+  // Copy from above. Add blocklength in sampling_options().
```

Comment:
It's better to avoid duplicated code. Try to call existing function instead of copying it. e.g., for this one, you can call another constructor from one constructor (https://en.cppreference.com/w/cpp/language/constructor#Delegating_constructor) and then add the additional code for blocklength.